spec/models/user_access_spec.rb

```diff
@@ -528,20 +238,538 @@
+    let!(:facility_4) { create(:facility, facility_group: facility_group_3_2) }
+    let!(:facility_5) { create(:facility) }
+    let!(:facility_6) { create(:facility) }
+
+    let!(:user_1) { create(:user, :with_phone_number_authentication, registration_facility: facility_1) }
+    let!(:user_2) { create(:user, :with_phone_number_authentication, registration_facility: facility_2) }
+    let!(:user_3) { create(:user, :with_phone_number_authentication, registration_facility: facility_3) }
+    let!(:user_4) { create(:user, :with_phone_number_authentication, registration_facility: facility_4) }
+    let!(:user_5) { create(:user, :with_phone_number_authentication, registration_facility: facility_5) }
+
+    let!(:admin_1) { create(:admin, :call_center, organization: organization_1) }
+    let!(:admin_2) { create(:admin, :call_center, organization: organization_1) }
+    let!(:admin_3) { create(:admin, :call_center, organization: organization_3) }
+    let!(:admin_4) { create(:admin, :call_center, organization: organization_3) }
+    let!(:admin_5) { create(:admin, :call_center) }
+
+    let!(:manager) { create(:admin, :manager) }
+    let!(:viewer_all) { create(:admin, :viewer_all) }
+    let!(:viewer_reports_only) { create(:admin, :viewer_reports_only) }
+    let!(:call_center) { create(:admin, :call_center) }
+
+    context "non power users" do
+      context "#accessible_organizations" do
+        let!(:permission_matrix) {
+          [
+            [manager, :manage, organization_3, [organization_3]],
+            [manager, :view_pii, organization_3, [organization_3]],
+            [manager, :view_reports, organization_3, [organization_3]],
+            [manager, :manage_overdue_list, organization_3, [organization_3]],
+
+            [viewer_all, :manage, organization_3, []],
+            [viewer_all, :view_pii, organization_3, [organization_3]],
+            [viewer_all, :view_reports, organization_3, [organization_3]],
+            [viewer_all, :manage_overdue_list, organization_3, [organization_3]],
+
+            [viewer_reports_only, :manage, organization_3, []],
+            [viewer_reports_only, :view_pii, organization_3, []],
+            [viewer_reports_only, :view_reports, organization_3, [organization_3]],
+            [viewer_reports_only, :manage_overdue_list, organization_3, []],
+
+            [call_center, :manage, organization_3, []],
+            [call_center, :view_pii, organization_3, []],
+            [call_center, :view_reports, organization_3, []],
+            [call_center, :manage_overdue_list, organization_3, [organization_3]]
+          ]
+        }
+
+        it "returns the organizations an admin can perform actions on" do
+          permission_matrix.each do |admin, action, access_resource, resources|
+            admin.accesses.create(resource: access_resource)
+            expect(admin.accessible_organizations(action)).to match_array(resources)
```

Comment:
@vkrmis Typically for table-tests, it's valuable to log[1] aggressively if a case fails, and generally provide lots of information about the test case[2]. Can you post some output on what happens if any of these cases break?

If it's not clear let's add some helpful logging.

[1] https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test
[2] https://github.com/simpledotorg/simple-server/blob/master/spec/models/user_spec.rb#L128
In this scenario, the `it` blocks have the case embedded in them, so it's clear what is running and what would've broken.

Reply:
Considered this, the problem is with the data for these specs being inaccessible outside the `it` block.

Adding custom logging as a workaround

Reply:
With the custom logging you'll see something like this:
```
access_level: viewer_reports_only
action: view_reports
expected: [#<FacilityGroup id: "04a9ab02-3c27-4900-b2ab-d225070466c7", name: "Kondagaon", description: "Visionary optimizing hardware", organization_id: "9dd7cca8-7f11-4bb6-a414-d687b312696f", created_at: "2020-09-08 12:52:50", updated_at: "2020-09-08 12:52:50", deleted_at: nil, protocol_id: "1145c716-51af-4f66-b40e-25e5d2960449", slug: "kondagaon_3", enable_diabetes_management: nil>, #<FacilityGroup id: "404ccaab-6ecc-405e-b905-028afc337ef5", name: "Bishwanath", description: "Synchronised 4th generation hierarchy", organization_id: "9dd7cca8-7f11-4bb6-a414-d687b312696f", created_at: "2020-09-08 12:52:50", updated_at: "2020-09-08 12:52:50", deleted_at: nil, protocol_id: "196ff98d-a35c-490b-b52f-6bbd4ab13a1c", slug: "bishwanath_4", enable_diabetes_management: nil>, #<Facility id: "63f1331b-fefa-453e-a1c2-3bfef077f00b", name: "Facility 4", street_address: "4 Gandhi Road", village_or_colony: "Colony 4", district: "Bathinda", state: "Punjab", country: "India", pin: "123456", facility_type: "PHC", created_at: "2020-09-08 12:52:50", updated_at: "2020-09-08 12:52:50", latitude: nil, longitude: nil, deleted_at: nil, facility_group_id: "404ccaab-6ecc-405e-b905-028afc337ef5", slug: "facility_4_4", zone: "Block ABC", enable_diabetes_management: true, facility_size: "small", monthly_estimated_opd_load: 300, enable_teleconsultation: true, teleconsultation_phone_number: "7711288220", teleconsultation_isd_code: "+20", teleconsultation_phone_numbers: [{"isd_code"=>"+1-939", "phone_number"=>"1329533107"}], import: false, organization_name: nil, facility_group_name: nil>]
got: [#<FacilityGroup id: "04a9ab02-3c27-4900-b2ab-d225070466c7", name: "Kondagaon", description: "Visionary optimizing hardware", organization_id: "9dd7cca8-7f11-4bb6-a414-d687b312696f", created_at: "2020-09-08 12:52:50", updated_at: "2020-09-08 12:52:50", deleted_at: nil, protocol_id: "1145c716-51af-4f66-b40e-25e5d2960449", slug: "kondagaon_3", enable_diabetes_management: nil>, #<FacilityGroup id: "404ccaab-6ecc-405e-b905-028afc337ef5", name: "Bishwanath", description: "Synchronised 4th generation hierarchy", organization_id: "9dd7cca8-7f11-4bb6-a414-d687b312696f", created_at: "2020-09-08 12:52:50", updated_at: "2020-09-08 12:52:50", deleted_at: nil, protocol_id: "196ff98d-a35c-490b-b52f-6bbd4ab13a1c", slug: "bishwanath_4", enable_diabetes_management: nil>]
```